health-services/plan-service/src/main/java/digit/service/validator/PlanEmployeeAssignmentValidator.java

```diff
@@ -62,28 +61,55 @@ public void validateCreate(PlanEmployeeAssignmentRequest request) {
         PlanEmployeeAssignment planEmployeeAssignment = request.getPlanEmployeeAssignment();
         String rootTenantId = centralInstanceUtil.getStateLevelTenant(request.getPlanEmployeeAssignment().getTenantId());
         List<PlanConfiguration> planConfigurations = commonUtil.searchPlanConfigId(planEmployeeAssignment.getPlanConfigurationId(), rootTenantId);
-        EmployeeResponse employeeResponse = hrmsUtil.fetchHrmsData(request.getRequestInfo(), planEmployeeAssignment.getEmployeeId(), planEmployeeAssignment.getTenantId());
+        UserDetailResponse userDetailResponse = userUtil.fetchUserDetail(getUserSearchReq(request.getRequestInfo(), planEmployeeAssignment.getEmployeeId(), planEmployeeAssignment.getTenantId()));
 
         // Validate if a same assignment already exists
         validateEmployeeAssignmentExistence(request);
 
         // Validate if plan config id exists
         validatePlanConfigId(planConfigurations);
 
-        // Validate if employee exists against HRMS
-        validateEmployeeAgainstHRMS(employeeResponse);
+        // Validate if employee exists against User Service
+        validateEmployeeAgainstUserService(userDetailResponse);
 
-        // Validate role of employee against HRMS
-        validateRoleAgainstHRMS(planEmployeeAssignment, employeeResponse);
+        // Validate role of employee against User Service
+        validateRoleAgainstUserService(planEmployeeAssignment, userDetailResponse);
 
         // Validate if role of employee is a conflicting role
         validateRoleConflict(planEmployeeAssignment);
 
-        // Validate campaign id, employee jurisdiction and highest root jurisdiction in case of National role
+        // Validate campaign id, employee jurisdiction and highest root jurisdiction in case of Root role
         validateCampaignDetails(planConfigurations.get(0).getCampaignId(), rootTenantId, request);
 
     }
 
+    /**
+     * This method validates the provided roles of the employee against User Service
+     *
+     * @param planEmployeeAssignment The plan employee assignment provided in request
+     * @param userDetailResponse     The user detail response from user service for the provided employeeId
+     */
+    private void validateRoleAgainstUserService(PlanEmployeeAssignment planEmployeeAssignment, UserDetailResponse userDetailResponse) {
+        List<String> userRolesFromUserService = userDetailResponse.getUser().get(0).getRoles().stream()
```

Comment:
call validateEmployeeAgainstUserService() from here instead of calling it outside, since this is a dependent function